scrapy_poet/utils.py

```diff
@@ -22,8 +22,50 @@ def get_callback(request, spider):
 
 
 class DummyResponse(Response):
+    """This class is returned by the ``InjectionMiddleware`` when it detects
+    that the download could be skipped. It inherits from Scrapy ``Response``
+    and signals and stores the URL and references the original ``Request``.
 
-    def __init__(self, url, request=None):
+    If you want to skip downloads, you can type annotate your parse method
+    with this class.
+
+    .. code-block:: python
+
+        def parse(self, response: DummyResponse):
+            pass
+
+    If there's no Page Input that depends on a Scrapy ``Response``, the
+    ``InjectionMiddleware`` is going to skip download and provide a
+    ``DummyResponse`` to your parser instead.
+
+    If your ``PageObjectInputProvider`` doesn't need a request, you simply
+    don't need to list it as a dependency. But if you need, for example, the
+    original request's URL, you can use ``DummyResponse`` instead of
+    ``Response``:
+
+    .. code-block:: python
+
+        @provides(ResponseData)
+        class ResponseDataProvider(PageObjectInputProvider):
+
+            def __init__(self, response: DummyResponse):
+                self.response = response
+
+            def __call__(self):
+                return ResponseData(
+                    url=self.response.url,
+                    html=self.get_data()
+                )
+
+            def get_data(self):
+                # make an api call
```

Comment:
to make an API call you would probably want an async def function

Reply:
I've updated this example to use async/await.